### ISSUE TITLE
[MAINT] Inferer.__call__() function in solaris.nets.infer.py does not properly use torch.no_grad()

### ISSUE THREAD

**Issue by imadtoubal** [CosmiQ/solaris#414]
## Maintenance request summary

Inferer.__call__() function in solaris.nets.infer.py does not properly use torch.no_grad(). PyTorch is effectively computing gradient for the inference process, thus, using significantly more GPU memory.

## Task detail and notes

Any additional information:

- This code change will not impact anything. It will only improve GPU memory efficiency.
